roles/channelserver/tasks/main.yml

```diff
@@ -0,0 +1,65 @@
+---
+
+- name: configure nginx upstream
+  become: true
+  template: src=upstream.conf.j2 dest=/etc/nginx/conf.d/upstream/http_channelserver.conf
+  notify: reload nginx config
+
+- name: configure nginx location
+  become: true
+  template: src=nginx.conf.j2 dest=/etc/nginx/conf.d/location/http_channelserver.conf
+  notify: reload nginx config
+
+- name: Pull channelserver docker image
+  become: true
+  docker_image:
+    state: present
+    name: rfkelly/channelserver
+    tag: "{{ channelserver_docker_tag }}"
+  register: image
+
+- debug: var=image
+
+- name: Create directory for downloading geolocation database
+  file: path=/data/config/channelserver/mmdb state=directory owner=app group=app mode=0775
+  become: true
+
+- name: Download geolocation database
+  become: true
+  get_url:
+    url: http://geolite.maxmind.com/download/geoip/database/GeoLite2-City.tar.gz
+    dest: /data/config/channelserver/mmdb/GeoLite2-City.tar.gz
```

Comment:
We might be able to arrange for this to periodically download a fresh copy of the geodb as part of the ansible cronjob, but I think it just grabs a static copy for now.